doc/system-telemetry/reboot-cause.md

```diff
@@ -0,0 +1,150 @@
+# Reboot-cause information via telemetry agent
+
+## Revision
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 09/02/20 | Sujin Kang  | Initial version    |
+
+## Scope
+-Enable sonic streaming telemetry agent to send Reboot-cause information
+
+### Enable sonic streaming telemetry agent to send Reboot-cause information
+
+#### Part 1
+During the boot, the `determine-reboot-cause` service ( previously `process-reboot-cause`) determines the last reboot-cause based on the hardware reboot-cause
+and the software reboot-cause information and `determine-reboot-cause` service will save the formatted last previous
+reboot cause information to "/host/reboot-cause/history/" with adding timestamp at the end of file name.
+`determine-reboot-cause` also will also create a symbolic link of the last reboot cause file to "/host/reboot-cause/previous-reboot-cause.json"
+
+The example shows the previous reboot-cause files stored in /host/reboot-cause/history/.
+```
+admin@sonic:~$ ls /host/reboot-cause/history/
+reboot-cause-2020_10_09_01_56_59.json
+reboot-cause-2020_10_09_02_00_53.json
+reboot-cause-2020_10_09_02_33_06.json
+reboot-cause-2020_10_09_04_53_58.json
+...
+```
+
+The following example shows the content of the previous reboot-cause file - reboot-cause-2020_10_09_04_53_58.json.
+```
+admin@sonic:~$ sudo cat /host/reboot-cause/history/reboot-cause-2020_10_09_04_53_58.json
+{"comment": "", "gen_time": "2020_10_09_04_53_58", "cause": "warm-reboot", "user": "admin", "time": "Fri Oct  9 04:51:47 UTC 2020"}
+```
+```
+[
+    {
+        gen_time: "2020_10_09_04_53_58",
+        cause: "warm-reboot",
+        user: "admin",
+        time: "Fri Oct  9 04:51:47 UTC 2020"
+        comment: ""
+    }
+]
+```
+
+#### Part 2
+A new service named as `process-reboot-cause.service` which will retrieve the saved reboot-cause files and read each reboot-cause information from the files
+and save the reboot-cause information up to 10 entries to state-DB.
+Verify the information from state-DB data is available via the cli command `show reboot-cause history` which is extended from `show reboot-cause`.
```

Comment:
Are we sticking with `show reboot-cause and `show reboot-cause history` or going with `show reboot cause` and `show reboot history`?

Reply:
`show reboot-cause` and `show reboot-cause history`